Packs/Zoom/Integrations/Zoom/Zoom.py

```diff
@@ -615,6 +831,740 @@ def check_authentication_type_parameters(api_key: str, api_secret: str,
+    client.zoom_update_channel(url_suffix, json_data)
+
+    return CommandResults(
+        readable_output=f"Channel {channel_id} was updated successfully",
+    )
+
+
+def zoom_invite_to_channel_command(client, **args) -> CommandResults:
+    """
+        invite users to Zoom channel
+    """
+    channel_id = args.get('channel_id')
+    user_id = args.get('user_id')
+    members = argToList(args.get('members'))
+    url_suffix = f'/chat/users/{user_id}/channels/{channel_id}/members'
+    json_members = [{"email": email} for email in members]
+
+    if user_id and re.match(emailRegex, user_id):
+        user_id = zoom_get_user_id_by_email(client, user_id)
+
+    members_json = {'members': json_members}
+
+    raw_data = client.zoom_invite_to_channel(members_json, url_suffix)
+
+    outputs = [{
+        'User id': raw_data.get('ids'),
+        'Channel ID': channel_id,
+        'Added at date and time': raw_data.get('added_at')
+    }]
+
+    human_readable = tableToMarkdown('Channel details', outputs, removeNull=True)
+
+    return CommandResults(
+        outputs_prefix='Zoom.Channel',
+        outputs_key_field='id',
+        readable_output=human_readable,
+        outputs=raw_data,
+        raw_response=raw_data
+    )
+
+
+def zoom_remove_from_channel_command(client, **args) -> CommandResults:
+    """
+        Remove a user from Zoom channel
+    """
+    client = client
+    channel_id = args.get('channel_id')
+    member_id = args.get('member_id')
+    user_id = args.get('user_id')
+
+    if user_id and re.match(emailRegex, user_id):
+        user_id = zoom_get_user_id_by_email(client, user_id)
+
+    url_suffix = f'/chat/users/{user_id}/channels/{channel_id}/members/{member_id}'
+    client.zoom_remove_from_channel(url_suffix)
+    return CommandResults(
+        readable_output=f'Member {member_id} was successfully removed from channel {channel_id}',
+    )
+
+
+def zoom_send_file_command(client, **args) -> CommandResults:
+    """
+        Send file in Zoom
+    """
+    client = client
+    user_id = args.get('user_id')
+    to_channel = args.get('to_channel')
+    to_contact = args.get('to_contact')
+    entry_id = args.get('entry_id')
+
+    if user_id and re.match(emailRegex, user_id):
+        user_id = zoom_get_user_id_by_email(client, user_id)
+
+    if "to_contact" not in args and "to_channel" not in args:
+        raise DemistoException(MISSING_ARGUMENT)
+
+    file_info = demisto.getFilePath(entry_id)
+    upload_url = f'https://file.zoom.us/v2/chat/users/{user_id}/messages/files'
+    json_data = remove_None_values_from_dict({
+        'to_contact': to_contact,
+        'to_channel': to_channel})
+    upload_response = client.zoom_send_file(upload_url, file_info, json_data)
+
+    message_id = upload_response.get('id')
+    return CommandResults(
+        readable_output=f'Message with  id {message_id} was  successfully sent'
+    )
+
+
+def parse_markdown_message(markdown_message: str, at_contact: str = None):
```

Comment:
I have 4 separated points about this function.
1. please add a detailed doctoring that explains it purpose.
2. I wonder if it is possible to split the different sections that already exist in the function to different functions(like compiler passes) that will make it shorter and readable?
3. do you know what is the cover of this function? I saw the UT, but considering the amount of cases I think we should have more flows, with distinct tests.
4. with relation to 3, I think we should have validations in the functions to the limits of the markdown(like amount of different mentions, or different styles in the smae message(not sure), and to return error when we get them. these cases should also have specific tests.

Reply:
added docstring  and split the function.
i have unit test that have most of the markdown option. i will add more.